docs/rfcs/021-autoscaling-compute-cache.md

```diff
@@ -0,0 +1,201 @@
+  extension.
+  
+- A new governor in the host that chooses which VM to allocate
+  how much memory.
+
+Picture:
+
+    +----------------------------+
+    |      Host                  |
+    | +------------------------+ |
+    | |    Container VM        | |               +------------+
+    | | +--------------------+ | |               |            |
+    | | |  Postgres          | | |               | Pageserver |
+    | | |                    | | |               |            |
+    | | |  ################  | | |               |            |
+    | | |  # shared_      #  | | |               +------------+
+    | | |  # buffers      #  | | |
+    | | |  #              #  | | |
+    | | |  .    |         .  | | |
+    | | |  .    |         .  | | |
+    | | |  .    V         .  | | |
+    | | |  ################  | | |
+    | | |                    | | |
+    | | +--------------------+ | |
+    | +------------------------+ |
+    +----------------------------+
+
+Pros:
+
+- best possible performance for the cached data
+
+Cons:
+
+- Scales only memory, cannot take advantage of local SSDs in host machine
+- Needs explicit operations to scale. Won't dynamicaly share resources
+  between tenants, we'll need to start a resizing process to change
+  the allocations.
+- Needs patches to core PostgreSQL
+
+
+# Alternative 2: Local filesystem cache
+
+Add code to Postgres Neon extension to use a local file on disk for
+caching.  When a page is evicted from Postgres buffer cache, write it
+to the local file, and read it back if it's requested again. Rely on
+kernel page cache to keep the most hot part of that file in memory.
+
+Like in solution 1, need a governor in the host to allocate the local
+disk for each VM, and orchestration to scale it up and down.
+
+
+    +----------------------------+
+    |      Host                  |
+    | +------------------------+ |
+    | |    Container VM        | |               +------------+
+    | | +--------------------+ | |               |            |
+    | | |  Postgres          | | |               | Pageserver |
+    | | |                    | | |               |            |
+    | | |  ################  | | |               |            |
+    | | |  # shared_      #  | | |               +------------+
+    | | |  # buffers      #  | | |
+    | | |  ################  | | |
+    | | |                    | | |
+    | | |  ################  | | |
+    | | |  # Local FS     #  | | |
+    | | |  # cache        #  | | |
+    | | |  #              #  | | |
+    | | |  ################  | | |
+    | | |                    | | |
+    | | +--------------------+ | |
+    | +------------------------+ |
+    +----------------------------+
+
+Pros:
+
+- No PostgreSQL core changes required
+- Automatically takes advantage of local SSDs, not just memory
+
+Cons:
+
+- Needs explicit operations to change the size of the cache file
+- Need support for live migration of the filesystem
```

Comment:
I think we need live migration of the filesystem even without this. In theory NFS is possible instead, but we've already ruled that out as being too slow.

Reply:
NFS has awful latencies. It can not be used neither for local file cache, neither for temporary postgres storage.